File: Gold/G5_12904.py
# A와 B, 2023년 11월 29일 19:30:39, 31252kb, 40ms, 210b
import sys
input = sys.stdin.readline

S=list(input().rstrip())
T=list(input().rstrip())

while len(S)!=len(T):
    if T[-1]=='A':
        T.pop()
    else:
        T.pop()
        T.reverse()

print(int(S==T))

File: Gold/G5_12904.java
# A와 B, 2023년 11월 29일 22:19:43, 14236kb, 140ms, 650b
import java.io.*;

public class Main {
	static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
	static StringBuilder S,T;
	static int sLen, tLen;
	
	public static void main(String args[]) throws IOException {
		S=new StringBuilder(input.readLine());
		T=new StringBuilder(input.readLine());
		
		sLen=S.length();
		tLen=T.length();
		
		game();
	}
	
	public static void game() {
		while(sLen!=tLen--) {
			if(T.charAt(tLen)=='A') {
				T.deleteCharAt(tLen);
			}else {
				T.deleteCharAt(tLen);
				T.reverse();
			}
		}
		if(T.toString().equals(S.toString())) System.out.println(1);
		else 			System.out.println(0);
	}
}
